cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -51,7 +51,7 @@ class MappingStep(CCIDictModel):
     "Step in a load or extract process"
     sf_object: str
     table: Optional[str] = None
-    fields_: Dict[str, str] = Field(..., alias="fields")
+    fields_: Optional[Union[Dict[str, str], List[str]]] = Field({}, alias="fields")
```

Comment:
I'm not sure. @prescod do you know the order of operations between applying validators and checking types?

Reply:
I think you can just try removing it from the type, and if this code is tested and typeguard doesn't complain, it's okay (which I _suspect_ is the case since I think validation happens in the process of constructing a model instance)

Reply:
I tested removing it from the type and immediately got typing failures, so it looks like type checking is applied prior to validators running.

Reply:
@davidmreed I just took a closer look. By adding `pre=True` to the validator decorator we can make it happen before other validation, and then it's possible to remove List from the type. I've added a commit with this change.